iroha/src/sumeragi.rs

```diff
@@ -302,6 +308,9 @@ impl Sumeragi {
         &mut self,
         no_tx_receipt: NoTransactionReceiptReceived,
     ) -> Result<(), String> {
+        if no_tx_receipt.leader_id != *self.network_topology.leader() {
+            return Ok(());
+        }
```

Comment:
I think this condition should be refactored and maybe moved to another place

Reply:
https://jira.hyperledger.org/browse/IR-809